### PR TITLE
Create dm-compatibility-catalog.md

### PR DESCRIPTION
**Pull request by sunzhaoyang**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

增加对上下游数据库的支持说明。

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.0 (TiDB 6.0 versions)
- [x] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: dm/dm-compatibility-catalog.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+- **Incompatible**。 不兼容的，指已发现明确存在无法兼容的情况，建议不要在生产环境中使用。
+
+## 数据源
+
+|数据源|级别|
+|-|-|
+|MySQL ≤ 5.5|Unknown|
+|MySQL 5.6|GA|
+|MySQL 5.7|GA|
+|MySQL 8.0|GA|
```
Comment (glorv) [inline dm/dm-compatibility-catalog.md:22]: 和 mysql 8.0 的兼容性应该是 Experimental 吧

File: dm/dm-compatibility-catalog.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+|-|-|
+|TiDB 3.x|GA|
+|TiDB 4.x|GA|
+|TiDB 5.0|GA|
+|TiDB 5.1|GA|
+|TiDB 5.2|GA|
+|TiDB 5.3|GA|
+|TiDB 5.4|GA|
+|MySQL|Unknown|
+|MariaDB|Unknown|
```
Comment (glorv) [inline dm/dm-compatibility-catalog.md:39]:
```suggestion
|MySQL|Experimental|
|MariaDB|Experimental|
```

File: dm/dm-compatibility-catalog.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
+---
+title: TiDB Data Migration 兼容性目录
+summary: 了解 DM 各版本与上下游各类型数据库的兼容关系
+---
+
+# TiDB Data Migration 兼容性目录
+
+DM 数据同步软件支持从不同类型的数据源迁移到 TiDB 集群。针对各种类型，产品支持程度可以分为三个级别：
```
Comment (lichunzhu) [inline dm/dm-compatibility-catalog.md:8]:
```suggestion
DM 数据同步软件支持从不同类型的数据源迁移到 TiDB 集群。针对各种数据源类型，产品支持程度可以分为三个级别：
```

File: dm/dm-compatibility-catalog.md (hunk 1 above)
Comment (lichunzhu) [reply dm/dm-compatibility-catalog.md:22]: @sunzhaoyang 啥标准算 GA 🤔

File: dm/dm-compatibility-catalog.md (hunk 2 above)
Comment (lichunzhu) [reply dm/dm-compatibility-catalog.md:39]: +1

File: dm/dm-compatibility-catalog.md (hunk 1 above)
Comment (sunzhaoyang) [reply dm/dm-compatibility-catalog.md:22]: 能让用户放心用的，含
1. 我们经过大量测试的
2. 广泛使用的

File: dm/dm-compatibility-catalog.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+---
+
+# TiDB Data Migration 兼容性目录
+
+DM 数据同步软件支持从不同类型的数据源迁移到 TiDB 集群。针对各种数据源类型，产品支持程度可以分为四个级别：
+
+- **GA**。正式支持，指该场景经过验证，并且通过完整的测试流程；
+- **Experimental**。实验性支持，即虽然通过部分验证，但测试尚未覆盖所有预设场景或使用者较少，存在少量场景下可能出错的风险；
+- **Unknown**。未知，DM 在迭代过程中尽量保证 MySQL 协议的兼容性，但由于资源限制，无法测试所有 MySQL 衍生版本。因此虽然技术原理上是兼容的，但是并未经完整测试，因此需要使用前自行验证；
+- **Incompatible**。 不兼容的，指已发现明确存在尚未兼容的情况，建议不要在生产环境中使用。
```
Comment (shichun-0415) [inline dm/dm-compatibility-catalog.md:13]:
```suggestion
- **Experimental**。实验性支持，即虽然通过部分验证，但测试尚未覆盖所有预设场景或用户较少，存在少量场景下可能出错的风险；
- **Unknown**。未知，DM 在迭代过程中尽量保证 MySQL 协议的兼容性，但由于资源限制，无法测试所有 MySQL 衍生版本。因此虽然技术原理上兼容，但是并未经完整测试，需要使用前自行验证；
- **Incompatible**。 不兼容，已发现明确不兼容的情况，不建议在生产环境中使用。
```

File: dm/dm-compatibility-catalog.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+## 数据源
+
+|数据源|级别|备注|
+|-|-|-|
+|MySQL ≤ 5.5|Unknown|
+|MySQL 5.6|GA|
+|MySQL 5.7|GA|
+|MySQL 8.0|Experimental|
+|MariaDB ＜ 10.1.2|Incompatible|时间类型的 binlog 不兼容|
```
Comment (shichun-0415) [inline dm/dm-compatibility-catalog.md:23]:
```suggestion
|MariaDB < 10.1.2|Incompatible|时间类型的 binlog 不兼容|
```

**Review by shichun-0415 — APPROVED** [review]
LGTM

File: dm/dm-compatibility-catalog.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+- **Incompatible**。 不兼容，已发现明确不兼容的情况，不建议在生产环境中使用。
+
+## 数据源
+
+|数据源|级别|备注|
+|-|-|-|
+|MySQL ≤ 5.5|Unknown|
+|MySQL 5.6|GA|
+|MySQL 5.7|GA|
+|MySQL 8.0|Experimental|
```
Comment (shichun-0415) [inline dm/dm-compatibility-catalog.md:22]:
```suggestion
|MySQL 5.6|GA||
|MySQL 5.7|GA||
|MySQL 8.0|Experimental||
```

File: dm/dm-compatibility-catalog.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+|MariaDB < 10.1.2|Incompatible|时间类型的 binlog 不兼容|
+|MariaDB 10.1.2 ~ 10.5.10|Experimental|
```
Comment (shichun-0415) [inline dm/dm-compatibility-catalog.md:24]:
```suggestion
|MariaDB 10.1.2 ~ 10.5.10|Experimental||
```

File: dm/dm-compatibility-catalog.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+|TiDB 6.0|GA|建议 DM ≥ 6.0|
+|TiDB 5.4|GA|建议 DM ≥ 5.4|
+|TiDB 5.3|GA|建议 DM ≥ 5.3|
+|TiDB 5.2|GA|建议 DM ≥ 2.0.7|
+|TiDB 5.1|GA|建议 DM ≥ 2.0.4|
+|TiDB 5.0|GA|建议 DM ≥ 2.0.4|
+|TiDB 4.x|GA|建议 DM ≥ 2.0.1|
+|TiDB 3.x|GA|建议 DM ≥ 2.0.1|
+|MySQL|Experimental|
+|MariaDB|Experimental|
```
Comment (shichun-0415) [inline dm/dm-compatibility-catalog.md:40]:
```suggestion
|MySQL|Experimental||
|MariaDB|Experimental||
```

File: dm/dm-compatibility-catalog.md [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+|MariaDB 10.1.2 ~ 10.5.10|Experimental|
+|MariaDB > 10.5.10|Incompatible|检查环节存在权限报错|
+
+## 目标数据库
+
+|数据源|级别|备注|
+|-|-|-|
+|TiDB 6.0|GA|建议 DM ≥ 6.0|
+|TiDB 5.4|GA|建议 DM ≥ 5.4|
+|TiDB 5.3|GA|建议 DM ≥ 5.3|
```
Comment (glorv) [inline dm/dm-compatibility-catalog.md:33]: 这个对 dm 版本选择的说明不是很清晰，比如用户具体应该选择什么版本，以及我们是否需要排除哪些有已知 critical 问题的版本。比如看到这个文档，我会认为对于 tidb v4.x, 我被建议使用 2.0.1， 但显然 v2.0.7+ 应该更好的选择。

**Review by lance6716 — APPROVED** [review]

File: dm/dm-compatibility-catalog.md (hunk 9 above)
Comment (sunzhaoyang) [reply dm/dm-compatibility-catalog.md:33]: 唔，本意是可用的最低版本，一般应该建议是用高版本。我再想想怎么表达

**Review by glorv — APPROVED** [review]
LGTM

File: dm/dm-compatibility-catalog.md [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+|MariaDB 10.1.2 ~ 10.5.10|实验支持||
+|MariaDB > 10.5.10|不兼容|检查环节存在权限报错|
+
+## 目标数据库
+
+> **警告：**
+>
+> 不建议使用 DM 5.3.0，因为当使用 GTID 同步且未开启 Relay log 的情况下，低概率会出现数据不同步。
+
+|数据源|级别|备注|
```
Comment (shichun-0415) [inline dm/dm-compatibility-catalog.md:33]:
```suggestion
|数据源|级别|DM 版本|
```

**Pull request merged**